### DISCUSSION TITLE
question about hive.allow-register-partition-procedure

### DISCUSSION THREAD

**Discussion by DongSeungLee** [trinodb/trino#21335] [Q&A] [answered]
i have a question, why allow-register-partition-procedure config is false as default, otherwise
unregister single partition is available.

The official document says that security problem might occur.
What types of security problem can occur?

you can see related PR(allow-register-partition-procedure)
https://github.com/trinodb/trino/commit/d14ed687299111aa99eded57673f5fc59e8ae59d

**Comment by mosabua** [comment]
I assume because the location already exists and presumably contains data, and that can potentially lead to issues like data corruption when write operations are performed. But I am mostly guessing - @findepi @kokosing might have more info

**Comment by findepi** [comment, marked as answer]
> I assume because the location already exists and presumably contains data

this can lead to inadvertent data exposure (unprivileged data access)

**Reply by DongSeungLee** [reply]
let me guess, you're saying that if someone does not know the details of HDFS(s3) path, using register partition procedure, someone can deduce the structure of stored data.
am i correct?

**Reply by hashhar** [reply]
Someone can register the partition in their own table which points to data from some other table that they don't have access to. So even if user can't query the table directly they can create their own table and add other tables partitions to it and query the data that way.

**Reply by DongSeungLee** [reply]
i appreciate your answer. It is very helpful!

**Comment by lozbrown** [comment]
@hashhar  @findepi wouldn't URL authorization similar to HiveServer2 be the answer here? IE granular control of locations users can/can't create tables/partitions
